test/lit/passes/optimize-instructions-gc-tnh.wast

```diff
@@ -48,43 +48,34 @@
     )
   )
 
-  ;; TNH:      (func $ref.eq-no (type $eqref_eqref_=>_none) (param $a eqref) (param $b eqref)
+  ;; TNH:      (func $ref.eq-no (type $eqref_eqref_anyref_=>_none) (param $a eqref) (param $b eqref) (param $any anyref)
   ;; TNH-NEXT:  (drop
   ;; TNH-NEXT:   (i32.const 1)
   ;; TNH-NEXT:  )
   ;; TNH-NEXT: )
-  ;; NO_TNH:      (func $ref.eq-no (type $eqref_eqref_=>_none) (param $a eqref) (param $b eqref)
+  ;; NO_TNH:      (func $ref.eq-no (type $eqref_eqref_anyref_=>_none) (param $a eqref) (param $b eqref) (param $any anyref)
   ;; NO_TNH-NEXT:  (drop
   ;; NO_TNH-NEXT:   (ref.eq
-  ;; NO_TNH-NEXT:    (block (result (ref $struct))
-  ;; NO_TNH-NEXT:     (drop
-  ;; NO_TNH-NEXT:      (ref.func $ref.eq-no)
-  ;; NO_TNH-NEXT:     )
-  ;; NO_TNH-NEXT:     (unreachable)
+  ;; NO_TNH-NEXT:    (ref.cast null $struct
+  ;; NO_TNH-NEXT:     (local.get $any)
   ;; NO_TNH-NEXT:    )
-  ;; NO_TNH-NEXT:    (block (result (ref data))
-  ;; NO_TNH-NEXT:     (drop
-  ;; NO_TNH-NEXT:      (ref.func $ref.eq-no)
-  ;; NO_TNH-NEXT:     )
-  ;; NO_TNH-NEXT:     (unreachable)
+  ;; NO_TNH-NEXT:    (ref.as_data
+  ;; NO_TNH-NEXT:     (local.get $any)
```

Comment:
We could only prove the cast would trap because it was casting a func to a struct. Now that's a validation error instead, so I had to change the test to use another internal reference type (i.e. a subtype of `any`)  instead of a func. The test needs an input type that is not `eq` and  the only internal reference that is not an `eq` is `any` itself. The pass can no longer prove that the cast traps, but at least it still shows that it keeps the input to `ref.eq` as `eq`, which is the intention.